app/Console/UpdateBookingAvailabilityCommand.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Silex\Console;
+
+use Broadway\CommandHandling\CommandBus;
+use CultuurNet\UDB3\Offer\Commands\UpdateBookingAvailability;
+use CultuurNet\UDB3\Offer\ValueObjects\BookingAvailability;
+use CultuurNet\UDB3\Offer\ValueObjects\BookingAvailabilityType;
+use CultuurNet\UDB3\Search\ResultsGenerator;
+use CultuurNet\UDB3\Search\SearchServiceInterface;
+use Exception;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ChoiceQuestion;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+use Symfony\Component\Console\Question\Question;
+
+class UpdateBookingAvailabilityCommand extends AbstractCommand
+{
+    private ResultsGenerator $searchResultsGenerator;
+
+    public function __construct(CommandBus $commandBus, SearchServiceInterface $searchService)
+    {
+        $this->searchResultsGenerator = new ResultsGenerator(
+            $searchService,
+            ['created' => 'asc'],
+            100
+        );
+
+        $this->setName('event:booking-availability:update');
+        $this->setDescription('Update the booking availability on a batch of events resulting from a SAPI 3 query.');
+
+        parent::__construct($commandBus);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $sapiQuestion = new Question(
+            'Provide SAPI query for events to update (periodic and permanent events are excluded by default)' . PHP_EOL
+        );
+        $sapiQuery = $this->getHelper('question')->ask($input, $output, $sapiQuestion);
+        $sapiQuery .= ' AND (calendarType:single OR calendarType:multiple)';
```

Comment:
I think we need to wrap the original query inside brackets here to be safe, to prevent issues like we had in https://jira.uitdatabank.be/browse/III-3061

Reply:
Nice catch, will be fixed with https://jira.uitdatabank.be/browse/III-4228